### DISCUSSION TITLE
Football Players Tracking

### DISCUSSION THREAD

**Discussion by studawson** [roboflow/notebooks#21] [Q&A]
Hi.  Id like to track players from my own footage in the Google Colab.  Can you show me how I download my own footage into the code please.  Does the code generate xy data for each player?

**Comment by SkalskiP** [comment]
> Does the code generate xy data for each player?

It gives you x, y, width, height.

> Id like to track players from my own footage in the Google Colab.

What would be the source for your data? GDrive?

**Reply by studawson** [reply]
It is currently saved to my laptop but I can upload to GDrive.

**Reply by SkalskiP** [reply]
@studawson please upload them to GDrive, and follow this video: https://www.youtube.com/watch?v=6UnCrulz-fE&ab_channel=AdrianDolinay to connect your Colab to GDrive :)

**Reply by studawson** [reply]
OK thanks.  I have connected Colab to GDrive.  How do I get the video into the model? I have minimal experience of coding.

**Reply by SkalskiP** [reply]
Hi @studawson :) https://github.com/roboflow-ai/notebooks/blob/main/notebooks/how-to-track-football-players.ipynb contains a video inference example. All you need to do is change `SOURCE_VIDEO_PATH` value to the location of your video.

**Reply by studawson** [reply]
Hi Piotr, ok.  I assume I don't need to run every cell in the Colab Notebook? Thanks.

**Reply by SkalskiP** [reply]
You don't need to run every cell, but the order of cells is essential, so to be on the safe side it is usually a much better idea to run all of them.

**Reply by studawson** [reply]
Ok, I tried but I think not being able to download from Kaggle until the competition is cancelled is a problem although I saw your link to clips n the YouTube comments.  What do I do with these clips?

**Reply by studawson** [reply]
I ran each cell, some were ok, some gave me StopIteration message or other error messages.

**Reply by SkalskiP** [reply]
You can use the link to download the videos I downloaded from Kaggle. This way, you can execute the whole tutorial without any errors.

**Reply by studawson** [reply]
OK, great where do I paste the link so I can download the videos?

**Comment by SkalskiP** [comment]
You can use `wget` or `curl` to download straight from the link. Alternatively, you can manually download the file and upload it to GDrive and connect it the same way as here: https://github.com/roboflow-ai/notebooks/discussions/21#discussioncomment-4387030

**Reply by studawson** [reply]
I think I uploaded the video files but I am still having problems when running the code.

**Reply by SkalskiP** [reply]
Hi, @studawson! I'd need a bit more information. What sort of problems do you experience?

**Comment by studawson** [comment]
Do I need to run these cells?

<img width="277" alt="Kaggle" src="https://user-images.githubusercontent.com/32792310/207968638-a2ab2fb6-97fc-4814-9093-d08049dd56f6.png">

**Comment by studawson** [comment]
<img width="927" alt="error 1" src="https://user-images.githubusercontent.com/32792310/207971520-9baf2cf7-8553-4214-b4a3-4f8cc1b78d5d.png">

**Comment by studawson** [comment]
<img width="343" alt="error 3" src="https://user-images.githubusercontent.com/32792310/207971634-cc719895-44e9-42db-bc3c-25c1f43017b1.png">

**Comment by studawson** [comment]
<img width="365" alt="error 4" src="https://user-images.githubusercontent.com/32792310/207971660-d775bc2a-42f3-4261-bf88-bb9bcd54b243.png">

**Comment by studawson** [comment]
<img width="350" alt="error 5" src="https://user-images.githubusercontent.com/32792310/207971720-f2190ba2-c6b3-4a64-b78c-de85f9ac8924.png">

**Comment by studawson** [comment]
<img width="374" alt="error 6" src="https://user-images.githubusercontent.com/32792310/207971757-ad0447c5-195a-43a3-9222-55d02aec2a1b.png">

**Comment by studawson** [comment]
<img width="425" alt="error 2" src="https://user-images.githubusercontent.com/32792310/207972649-bd3eb37c-13b1-4ba1-99ff-ae01d745fdff.png">

**Comment by studawson** [comment]
Sorry to drop all this on you Piotr.  I have little experience of coding.  I appreciate all your help.

**Comment by studawson** [comment]
Here is the link to my copy of the Colab code.

https://colab.research.google.com/drive/1lvVCPjSSoOisNgkCE-BiZMlVcKi2efaM?usp=sharing

**Comment by studawson** [comment]
Hi Piotr.  Did you have a chance to look at the screenshots I sent?  Thanks.

**Comment by studawson** [comment]
Hi Piotr, I understand this won't work with my own footage. Is that right?
